### PR TITLE
basic cart functionality

### PR DESCRIPTION
**Pull request by jamesjamail**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -22,6 +21,19 @@ app.get('/getData/:itemId', (request, response) => {
     },request.params.itemId)
 })
 
+app.get('/search/:searchString', (request, response) => {
+    db.searchProducts((err, results) => {
+        if (err) {
+            console.log('GETPRODUCTDATAERROR: ', err);
```
Comment (RSP531) [inline server/server.js]: Do you want to response.send the error, or just console log it?

File: server/server.js [hunk 2]
```diff
@@ -22,6 +21,19 @@ app.get('/getData/:itemId', (request, response) => {
...
+        }
+        else {
+            response.send(results);
+            console.log('sending data from server');
+        }
+        response.end();
```
Comment (RSP531) [inline server/server.js]: do you want this response.end to still be in your search products query, or could you put it outside?

File: server/server.js (hunk 1 above)
Comment (jamesjamail) [reply server/server.js]: Nice, moved it outside the db method.

Comment (jamesjamail) [reply server/server.js]: Actually, moving it outside the db method causes the code to break

**Pull request merged**